libs/hdf-converters/src/neuvector-mapper.ts

```diff
@@ -0,0 +1,311 @@
+  digest: string;
+  cmds: string;
+  vulnerabilities: RESTVulnerability[];
+  size: number /* int64 */;
+};
+
+export type NeuvectorScanJson = {
+  report: RESTScanRepoReport;
+  error_message: string;
+};
+
+const CWE_NIST_MAPPING = new CweNistMapping();
+
+function cweTags(description: string): string[] | undefined {
+  const regex = /CWE-\d{3}/g;
+  return description?.match(regex)?.flat();
+}
+
+function nistTags(cweTags: string[] | undefined): string[] {
+  const identifiers =
+    cweTags?.map((tag: string) => tag.match(/\d{3}/g)?.[0] ?? [])?.flat() ?? [];
+  return CWE_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_UPDATE_REMEDIATION_NIST_TAGS
+  );
+}
+
+function ghsaTag(name: string): string | undefined {
+  const regex = /GHSA-[a-z|0-9]{4}-[a-z|0-9]{4}-[a-z|0-9]{4}/;
+  return regex.exec(name)?.[0];
+}
+
+function rhsaTag(name: string): string | undefined {
+  const regex = /RHSA-[a-z|0-9]{4}:[a-z|0-9]{4}/;
+  return regex.exec(name)?.[0];
+}
+
+function cveIdMatches(cveName: string): (value: RESTModuleCve) => boolean {
+  return (cve: RESTModuleCve) => cve.name === cveName;
+}
+
+function universalTags(): MappedTransform<
+  {[key: string]: any} & ILookupPath,
+  ILookupPath
+> {
+  // Heimdall currently doesn't have a way to display passthrough data, and this information would be useful to view per vulnerability.
+  return {
+    envs: {
+      path: '$.report.envs',
+      transformer: (envs: string[]) => (!envs ? undefined : envs)
+    },
+    cmds: {
+      path: '$.report.cmds',
+      transformer: (cmds: string[]) => (!cmds ? undefined : cmds)
+    }
+  };
+}
+
+export class NeuvectorMapper extends BaseConverter {
+  withRaw: boolean;
+  rawData: NeuvectorScanJson;
+  getModule: (moduleName: string) => RESTScanModule | undefined;
+
+  memoizedGetModule(): (moduleName: string) => RESTScanModule | undefined {
+    return (moduleName: string) => {
+      return this.rawData.report.modules?.find(
+        (value: RESTScanModule) => value.name === moduleName
+      );
+    };
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion,
+      target_id: null
+    },
+    version: HeimdallToolsVersion,
+    statistics: {
+      duration: null
+    },
+    profiles: [
+      {
+        name: 'Neuvector',
+        title: null,
+        version: null,
+        maintainer: null,
+        summary: null,
+        license: null,
+        copyright: null,
+        copyright_email: null,
+        supports: [],
+        attributes: [],
+        depends: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'report.vulnerabilities',
+            key: 'id',
+            tags: {
+              cves: {path: 'cves'},
+              cpes: {
+                path: 'cpes',
+                transformer: (cpes: string[]) => (!cpes ? undefined : cpes)
```

Comment:
https://github.com/mitre/heimdall2/blob/master/libs/hdf-converters/src/utils/global.ts#L82

if you want to conditionally supply attributes, you might be interested in leveraging this pre-existing helper function